### PR TITLE
rubocop - reduce complexity

### PR DESCRIPTION
**Pull request by Br3nda**

### CODE REVIEW

File: app/controllers/seeds_controller.rb [hunk 1]
```diff
@@ -109,4 +103,14 @@ def seed_params
       :days_until_maturity_min, :days_until_maturity_max, :organic, :gmo,
       :heirloom, :tradable_to, :slug)
   end
+
+  def seeds
+    if @owner
```
Comment (pozorvlak) [inline app/controllers/seeds_controller.rb]: I'd be tempted to further simplify this into

```
s =
  if @owner then @owner.seeds
  elsif @crop then @crop.seeds
  else Seed end
s.includes(:owner, :crop).paginate(page: params[:page])
```

File: app/models/harvest.rb [hunk 2]
```diff
@@ -82,32 +82,38 @@ def harvest_slug
...
-      string += "#{number_to_human(quantity.to_s, strip_insignificant_zeros: true)} "
-      string += if unit == 'individual'
-                  'individual '
-                elsif quantity == 1
-                  "#{unit} of "
-                else
-                  "#{unit.pluralize} of "
-                end
-    end
+    "#{quantity_to_human} #{unit_to_human} #{crop_name_to_human} #{weight_to_human}".strip
```
Comment (pozorvlak) [inline app/models/harvest.rb]: This is *much* better. Awesome!

File: app/models/member.rb [hunk 3]
```diff
@@ -221,17 +221,10 @@ def self.nearest_to(place)
...
-      newsletter_subscribe
-    elsif confirmed_at # i.e. after member's confirmed their account
-      if newsletter_changed? # edited member settings
-        if newsletter
-          newsletter_subscribe
-        else
-          newsletter_unsubscribe
-        end
-      end
+    if newsletter
```
Comment (pozorvlak) [inline app/models/member.rb]: This is shorter, but I find it a lot harder to follow than the previous version. Was the problem that rubocop was complaining about the if-clauses being nested too deeply? We could easily trim one level by changing L226 to `if confirmed_at && newsletter_changed?`.

File: app/models/member.rb [hunk 4]
```diff
@@ -221,17 +221,10 @@ def self.nearest_to(place)
...
+      newsletter_subscribe confirmed_at_changed? || confirmed_at && newsletter_changed?
     end
+    newsletter_unsubscribe if confirmed_at && newsletter_changed? && !newsletter
```
Comment (pozorvlak) [inline app/models/member.rb]: You could put this in an `else`-clause to the previous `if` and avoid checking `newsletter` again.

File: app/models/member.rb [hunk 5]
```diff
@@ -221,17 +221,10 @@ def self.nearest_to(place)
...
-    elsif confirmed_at # i.e. after member's confirmed their account
-      if newsletter_changed? # edited member settings
-        if newsletter
-          newsletter_subscribe
-        else
-          newsletter_unsubscribe
-        end
-      end
+    if newsletter
+      newsletter_subscribe confirmed_at_changed? || confirmed_at && newsletter_changed?
```
Comment (pozorvlak) [inline app/models/member.rb]: Should there be an `if` between `newsletter_subscribe` and `confirmed_at_changed`? The first argument of `newsletter_subscribe` shouldn't be a boolean.

I'd also prefer brackets in the condition to make the precedence clear.

File: app/models/planting.rb [hunk 6]
```diff
@@ -112,6 +112,20 @@ def planted?(current_date = Date.current)
     planted_at.present? && current_date.to_date >= planted_at
   end
 
+  def finished_at_date_passed?
```
Comment (pozorvlak) [inline app/models/planting.rb]: You're right, this code definitely belongs on the model.

File: app/models/member.rb (hunk 5 above)
Comment (Br3nda) [inline app/models/member.rb]: Wow. Yes. I'll add more tests too here

**Comment by Br3nda** [comment]
I'm gonna split these out into PRs

File: app/models/harvest.rb (hunk 2 above)
Comment (Br3nda) [inline app/models/harvest.rb]: I've moved this into it's own pr here  #1177

File: app/controllers/seeds_controller.rb (hunk 1 above)
Comment (Br3nda) [inline app/controllers/seeds_controller.rb]: i've done this, and removed the `s` variable -- it's on PR  #1176

**Comment by Br3nda** [comment]
I'm gonna rebase this -- hoping the comments don't get lost by github
